src/url-handling.js

```diff
@@ -90,14 +104,40 @@ export function urlStateToUrlObject(urlState: UrlState): UrlObject {
     range:
       stringifyRangeFilters(urlState.profileSpecific.rangeFilters) || undefined,
     thread: urlState.profileSpecific.selectedThread,
-    threadOrder: urlState.profileSpecific.threadOrder.join('-'),
+    globalTrackOrder: urlState.profileSpecific.globalTrackOrder.join('-'),
     file: urlState.pathInZipFile || undefined,
     v: CURRENT_URL_VERSION,
   };
 
-  // Add the parameter hiddenThreads only when needed
-  if (urlState.profileSpecific.hiddenThreads.length > 0) {
-    query.hiddenThreads = urlState.profileSpecific.hiddenThreads.join('-');
+  // Add the parameter hiddenGlobalTracks only when needed.
+  if (urlState.profileSpecific.hiddenGlobalTracks.size > 0) {
+    query.hiddenGlobalTracks = '';
+    for (const trackIndex of urlState.profileSpecific.hiddenGlobalTracks) {
+      query.hiddenGlobalTracks += `${trackIndex}-`;
+    }
+    // Slice off the trailing '-';
+    query.hiddenGlobalTracks = query.hiddenGlobalTracks.slice(0, -1);
+  }
+
+  let hiddenLocalTracksByPid = '';
+  for (const [pid, tracks] of urlState.profileSpecific.hiddenLocalTracksByPid) {
+    if (tracks.size > 0) {
+      hiddenLocalTracksByPid += [pid, ...tracks].join('-') + '~';
+    }
+  }
+  if (hiddenLocalTracksByPid.length > 0) {
+    // Ony add to the query string if something was actually hidden.
+    // Also, slice off the last '~'.
+    query.hiddenLocalTracksByPid = hiddenLocalTracksByPid.slice(0, -1);
+  }
+
+  query.localTrackOrderByPid = '';
+  for (const [pid, trackOrder] of urlState.profileSpecific
+    .localTrackOrderByPid) {
+    if (trackOrder.length > 0) {
+      query.localTrackOrderByPid +=
+        `${String(pid)}-` + trackOrder.join('-') + '~';
```

Comment:
previously you used something like `[pid, ...trackOrder].join('-')` but not here, any reason ?

Reply:
I guess I'm not consistent, although it all gets the job done.